src/Shared/ResultsHelpers/SharedUrlHelper.cs

```diff
@@ -32,49 +33,50 @@ internal static class SharedUrlHelper
 
     internal static bool IsLocalUrl([NotNullWhen(true)] string? url)
     {
-        if (string.IsNullOrEmpty(url))
+        var urlSpan = url.AsSpan();
+
+        if (urlSpan.IsEmpty)
         {
             return false;
         }
 
         // Allows "/" or "/foo" but not "//" or "/\".
-        if (url[0] == '/')
+        if (urlSpan[0] == '/')
         {
             // url is exactly "/"
-            if (url.Length == 1)
+            if (urlSpan.Length < 2)
             {
                 return true;
             }
 
             // url doesn't start with "//" or "/\"
-            if (url[1] != '/' && url[1] != '\\')
+            if (urlSpan[1] is not '/' and '\\')
```

Comment:
I really don't like the pattern matching syntax in these types of cases. This is wrong because `not` binds to the `'/'` check but not to the `'\\'` check.
https://learn.microsoft.com/dotnet/csharp/language-reference/operators/patterns#precedence-and-order-of-checking